debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/MongoDbConnectorIT.java

```diff
@@ -143,6 +144,96 @@ public void shouldFailToValidateInvalidConfiguration() {
         assertNoConfigurationErrors(result, CommonConnectorConfig.TOMBSTONES_ON_DELETE);
     }
 
+    @Test
+    public void shouldThrowExceptionWhenFieldExcludeListDatabasePartIsOnlyProvided() {
+        shouldValidateFilterFieldConfiguration(MongoDbConnectorConfig.FIELD_EXCLUDE_LIST, "inventory", 1);
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenFieldExcludeListDatabaseAndCollectionPartIsOnlyProvided() {
+        shouldValidateFilterFieldConfiguration(MongoDbConnectorConfig.FIELD_EXCLUDE_LIST, "inventory.collectionA", 1);
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenFieldExcludeListDatabaseAndCollectionPartsAreMissing() {
+        shouldValidateFilterFieldConfiguration(MongoDbConnectorConfig.FIELD_EXCLUDE_LIST, ".name", 1);
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenFieldExcludeListFieldPartIsMissing() {
+        shouldValidateFilterFieldConfiguration(MongoDbConnectorConfig.FIELD_EXCLUDE_LIST, "db1.collectionA.", 1);
+    }
+
+    @Test
+    public void shouldNotThrowExceptionWhenFieldExcludeListHasLeadingWhiteSpaces() {
+        shouldValidateFilterFieldConfiguration(MongoDbConnectorConfig.FIELD_EXCLUDE_LIST, " *.collectionA.name", 0);
```

Comment:
On all the "should not throw" methods, do we have corresponding tests in `FiltersTest` which assert that the filters actually work as intended, by asserting against matching and non matching field names? In particular, I'm wondering about the whitespace behavior: is this handled correctly both during validation and actual filter processing?

Reply:
@gunnarmorling I have added the respective test cases for all the "should not throw" methods in `FiltersTest` by doing a `NotNull` check for now. So to exactly match and assert how may I get the values out of the `FieldSelector`?

Reply:
Something like this should work:

```
        filters = build.excludeFields(" *.collectionA.key1").createFilters();
        assertEquals(
        		Document.parse(" { \"key1\" : \"value1\", \"key2\" : \"value2\" }"),
        		filters.fieldFilterFor(null).apply(Document.parse(" { \"key2\" : \"value2\" }"))
        );
```

Also the test names should be adjusted accordingly, i.e. not focusing on the fact of not throwing an exception but e.g. "excludeFilterShouldRemoveMatchingField()", etc.

Reply:
Thanks, @gunnarmorling, I have added the tests accordingly and adjusted the test names as well.